tools/json_tools/convert_to_portions.py

```diff
@@ -81,38 +38,50 @@ def gen_new(path):
     # Having troubles with this script halting?
     # Uncomment below to find the file it's halting on
     #print(path)
-    with open(path, "r") as json_file:
-        json_data = json.load(json_file)
-        for jo in json_data:
-            if "type" not in jo:
-                continue
-            if (jo["type"] != "ARMOR" and
-                jo["type"] != "TOOL_ARMOR" and
-               "armor_data" not in jo):
-                continue
-
-            if "armor_data" in jo:
-                new_data = portionize(jo["armor_data"])
-                if new_data is not None:
-                    jo["armor_data"] = new_data
+    try:
+        with open(path, "r") as json_file:
+            json_data = json.load(json_file)
+            for jo in json_data:
+                if type(jo) is not dict:
+                    print(f"Incorrectly formatted JSON data at {path}")
+                    print("Ensure that the file is a JSON file consisting"
+                          " of an array of objects!")
+                    break
+
+                if ("type" not in jo or
+                    jo["type"] != "ARMOR" and
+                    jo["type"] != "TOOL_ARMOR" and
+                   "armor_data" not in jo):
+                    continue
+
+                if "armor_data" in jo:
+                    new_data = portionize(jo["armor_data"])
+                    if new_data is not None:
+                        jo["armor_data"] = new_data
+                        change = True
+                    continue
+
+                # I don't know python - assiging to jo is not necessary here
+                # but is necessary for "armor_data" above
+                new_jo = portionize(jo)
+                if new_jo is not None:
                     change = True
-                continue
-
-            # I don't know python - assiging to jo is not necessary here
-            # but is necessary for "armor_data" above
-            new_jo = portionize(jo)
-            if new_jo is not None:
-                change = True
+    except json.JSONDecodeError:
+        print(f"Json Decode Error at {path}")
+        print("Ensure that the file is a JSON file consisting of"
+              " an array of objects!")
 
     return json_data if change else None
```

Comment:
This will throw a `NameError` if the `except` clausge gets hit because `json_data` is defined on the line that would throw this.